client/app/collections/upload_queue.coffee

```diff
@@ -210,13 +258,12 @@ module.exports = class UploadQueue extends Backbone.Collection
             unless dir = dirs[i++]
                 # folders will be created
                 # we can safely add files to bottom of queue
-                blobs = _.filter blobs, (blob) ->
-                    blob.name not in ['.', '..']
+                blobs = _.filter blobs, (blob) -> blob.name not in ['.', '..']
                 @addBlobs blobs, parent
                 return
 
             prefix = parent.getRepository()
-            parts = dir.split('/').filter (x) -> x # ?remove empty last part
+            parts = dir.split('/').filter (x) -> x # remove empty last part
```

Comment:
I didn't even understand it before trying hard this morning ahaha. It is very specific to folder's upload. I move it to heplers.